canaries/2D_image_stream.py

```diff
@@ -148,6 +145,7 @@ async def main():
         await close_robot(robot)
         logging.info("exiting with status 1")
         exit(1)
+    await close_robot(robot)
```

Comment:
This is the fix. The responsibility of closing the robot is left to the function that created it rather than sub-functions that may step on each other's toes.